py/dml/messages.py

```diff
@@ -1511,6 +1515,14 @@ class EVLACONST(DMLError):
     fmt = ("variable length array declared with (partially) const-qualified "
            + "type")
 
+class ESIZEOFOP(DMLError):
+    """
+    The operand of 'sizeof' contains an unknown identifier.
+    This often occurs because `sizeof` was used when `sizeoftype` was intended.
```

Comment:
This text appears in documentation and describes the error message kind, not the error. Current is OK, but you can consider smth like "This is a variant of the EIDENT message that..."

Reply:
Ok. Also renamed it to `EIDENTSIZEOF` to really rub it in.